### PR TITLE
Fix signum comparison for big numbers

### PR DESCRIPTION
**Pull request by martintreurnicht**
Right now `isPositive`, `isNegative`, `isZero`, `isNotZero` does not work for `BigInteger` or `BigDecimal`. I tried several things but what i settled on is just converting numbers to it's double value, this is actually totally fine in my opinion because we don't care about precision, we just want to get the actual signum 0, 1, -1. If there was a way to just get a signum of the abstract number class we'd be done, but unfortunately it's not that simple (as far as i know).

### CODE REVIEW

File: assertk-jvm/src/test/kotlin/test/assertk/assertions/support/NumberSpec.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import assertk.assertions.isPositive
+import assertk.assertions.isZero
+import org.junit.Test
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlin.test.assertEquals
+import kotlin.test.assertFails
+
+
+class BigDecimalSpecNumberOnIsZero {
```
Comment (evant) [inline assertk-jvm/src/test/kotlin/test/assertk/assertions/support/NumberSpec.kt]: Can you update the tests? should be able to just add them in `NumberTest`

Comment (martintreurnicht) [reply assertk-jvm/src/test/kotlin/test/assertk/assertions/support/NumberSpec.kt]: They can't be moved to `NumberTest`, that test is environment agnostic and doesn't support `BigDecimal` or `BigInteger`

Comment (evant) [reply assertk-jvm/src/test/kotlin/test/assertk/assertions/support/NumberSpec.kt]: In that case, make `JavaNumberTest`.

Comment (martintreurnicht) [reply assertk-jvm/src/test/kotlin/test/assertk/assertions/support/NumberSpec.kt]: Done

**Pull request merged**